File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int m=0;
        int minp=INT_MAX;
        for(int pr : prices)
        {
            
            minp=min(pr,minp);

            m=max(pr-minp,m);
            
        }
        return m;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        p=0
        b=float("inf")
        for i in range(len(prices)):
            b=min(b,prices[i])
            p=max(p,prices[i]-b)
        return p